lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/impl/DefaultRemoteManagerImplementation.java

```diff
@@ -95,15 +95,16 @@
 
     this.handlerContainer.setTransport(this.transport);
 
-    this.myIdentifier = new SocketRemoteIdentifier((InetSocketAddress)this.transport.getLocalAddress());
+    InetSocketAddress address = new InetSocketAddress(localAddressProvider.getLocalAddress(), this.transport.getListeningPort());
+    this.myIdentifier = new SocketRemoteIdentifier(address);
```

Comment:
this.transport.getLocalAddress() points to the hostname of the container. For communication purposes, that is what is needed. But this.myIdentifier is how the container announces itself to other containers. And so, we have to use the getlocalAddress() from the localAddressProvider.

When using containers, localAddressProvider is the ContainerBasedLocalAddressProvider. Otherwise, it is the HostnameBasedLocalAddressProvider. So this code change won't affect the non-container pools while making the minor tweak necessary for container pools.

---
In reply to: [196512374](https://github.com/sharathmalladi/reef/pull/5#discussion_r196512374) [](ancestors = 196512374)